signer/src/transaction_coordinator.rs

```diff
@@ -1672,76 +1695,295 @@ where
+        // - [7]  Is not older than the provided withdrawal expiry threshold.
+        //
+        // We set the context window for the pending-accepted query below to the
+        // number of blocks that the withdrawal request is considered
+        // valid (not expired). This limits the number of blocks the query will
+        // consider when fetching pending withdrawal requests.
         let pending_withdraw_requests = storage
             .get_pending_accepted_withdrawal_requests(
-                bitcoin_chain_tip.as_ref(),
-                stacks_chain_tip,
+                params.bitcoin_chain_tip.as_ref(),
+                params.stacks_chain_tip,
                 min_bitcoin_height,
-                accept_threshold,
+                params.signature_threshold,
             )
             .await?;
 
-        let mut deposits: Vec<utxo::DepositRequest> = Vec::new();
+        // If we didn't find any pending withdrawal requests, we can exit early.
+        if pending_withdraw_requests.is_empty() {
+            tracing::debug!("no pending withdrawal requests eligible for consideration found");
+            return Ok(eligible_withdrawals);
+        }
 
-        for req in pending_deposit_requests {
+        // Iterate over the pending withdrawal requests we fetched above and
+        // validate them against the remaining consensus rules.
+        for req in pending_withdraw_requests {
+            if req.bitcoin_block_height < min_soft_bitcoin_height {
+                tracing::debug!(
+                    request_id = req.request_id,
+                    bitcoin_block_height = req.bitcoin_block_height,
+                    min_soft_bitcoin_height,
+                    reason = SKIP_REASON_SOFT_EXPIRY,
+                    message = REQUEST_SKIPPED_MESSAGE
+                );
+                continue;
+            }
+
+            // [8] Ensure that the withdrawal request amount is above the dust
+            // limit.
+            if req.amount <= WITHDRAWAL_DUST_LIMIT {
+                tracing::debug!(
+                    request_id = req.request_id,
+                    amount = req.amount,
+                    reason = SKIP_REASON_AMOUNT_IS_DUST,
+                    message = REQUEST_SKIPPED_MESSAGE
+                );
+                continue;
+            }
+
+            // [9] Ensure that the withdrawal request amount is within the
+            // current sBTC caps.
+            let per_withdrawal_cap = params.sbtc_limits.per_withdrawal_cap().to_sat();
+            if req.amount > per_withdrawal_cap {
+                tracing::debug!(
+                    request_id = req.request_id,
+                    amount = req.amount,
+                    per_withdrawal_cap = per_withdrawal_cap,
+                    reason = SKIP_REASON_PER_WITHDRAWAL_CAP_EXCEEDED,
+                    message = REQUEST_SKIPPED_MESSAGE
+                );
+                continue;
+            }
```

Comment:
For deposits the cap check is done in `SbtcRequests`, should we do the same for withdrawals for consistency?

Reply:
mmm I don't know, but yeah agreed that this is something we can tackle at another time, for now it's enough to ensure we have all the rules in place 😅